src/wallet/api/wallet.cpp

```diff
@@ -85,11 +85,14 @@ struct Wallet2CallbackImpl : public tools::i_wallet2_callback
 
     virtual void on_new_block(uint64_t height, const cryptonote::block& block)
     {
-        //LOG_PRINT_L3(__FUNCTION__ << ": new block. height: " << height);
-
-        if (m_listener) {
-            m_listener->newBlock(height);
-            //  m_listener->updated();
+        // Don't flood the GUI with signals. On fast refresh - send signal every 1000th block
+        // get_refresh_from_block_height() returns the blockheight from when the wallet was 
+        // created or the restore height specified when wallet was recovered
+        if(height > m_wallet->m_wallet->get_refresh_from_block_height() || height % 1000 == 0) {
+            // LOG_PRINT_L3(__FUNCTION__ << ": new block. height: " << height);
+            if (m_listener) {
+                m_listener->newBlock(height);
+            }
```

Comment:
I think it should be >=. Not sure it really matters though, depends what hooks up on that signal. Also, when happens when it's in "normal being on the network waiting for blocks" mode ? Does it get a signal every 2000 minutes only ?